pkg/chunk/local/boltdb_index_client.go

```diff
@@ -285,31 +282,40 @@ func (b *BoltIndexClient) QueryDB(ctx context.Context, db *bbolt.DB, query chunk
 	})
 }
 
-type boltWriteBatch struct {
-	puts    map[string]map[string][]byte
-	deletes map[string]map[string]struct{}
+type TableWrites struct {
+	puts    map[string][]byte
+	deletes map[string]struct{}
 }
 
-func (b *boltWriteBatch) Delete(tableName, hashValue string, rangeValue []byte) {
-	table, ok := b.deletes[tableName]
+type BoltWriteBatch struct {
+	Writes map[string]TableWrites
+}
+
+func (b *BoltWriteBatch) getOrCreateTableWritesBatch(tableName string) TableWrites {
```

Comment:
I would name this `getOrCreateTableWrites` because the struct is called `TableWrites`, unless you intended to call the struct `TableWritesBatch` (in that case I would rename the struct).